rclcpp/include/rclcpp/executor.hpp

```diff
@@ -107,11 +109,13 @@ class Executor
     }
   }
 
-  void spin_node_once(rclcpp::node::Node::SharedPtr & node, bool nonblocking = false)
+  template<typename Rep = int64_t, typename Period = std::milli>
+  void spin_node_once(rclcpp::node::Node::SharedPtr & node, std::chrono::duration<Rep,
+    Period> timeout = std::chrono::duration<Rep, Period>::zero())
```

Comment:
I think it would be good to make this non-blocking by default, by setting the default value to something negative, in order to make it more obvious that this possible.